src/resolvers/deck/createDeck.ts

```diff
@@ -28,3 +35,190 @@ export const createDeck = mutationWithClientMutationId({
     return { deck }
   },
 })
+
+export const importDeck = mutationWithClientMutationId({
+  name: 'ImportDeck',
+  description: 'Import/Duplicate a deck entity',
+  inputFields: {
+    id: { type: GraphQLNonNull(GraphQLID), description: 'Public deck id' },
+  },
+  outputFields: {
+    deck: { type: DeckType, description: 'Imported deck' },
+  },
+  mutateAndGetPayload: async (
+    { id },
+    { user, publishedDeckLoader, modelLoader, fieldsByModelLoader }: Context
+  ) => {
+    const helperMap: {
+      updateModelIndex: { oldId: string; newId: string }[]
+      updateFieldIndex: { oldId: string; newId: string }[]
```

Comment:
why use an array for this? I believe if you had an map where the key was the old id and the value the new id it would simplify the logic to find the corresponding new id from an old id.
```suggestion
      updateModelIndex: Record<string, string>
      updateFieldIndex: Record<string, string>
```

Reply:
I agree this is a better way to solve it.